### PR TITLE
Feature: Add support for app manifest v3

### PR DESCRIPTION
**Pull request by doosuu**
PR adds support for app manifest v3 while still keeping backwards compatibility support for manifests < v3

### CODE REVIEW

File: vehicle-model-lifecycle/src/download_vspec.py [hunk 1]
```diff
@@ -84,21 +87,104 @@ def download_file(uri: str, local_file_path: str):
...
+def is_legacy_app_manifest(app_manifest_dict: Dict[str, Any]) -> bool:
+    """Check if the used app manifest is a legacy file.
+
+    Args:
+        app_manifest_dict (Dict[str, Any]): The app manifest.
+
+    Returns:
+        bool: True if the app manifest is a legacy file, False if not.
+    """
+    return "manifestVersion" not in app_manifest_dict
```
Comment (MP91) [inline vehicle-model-lifecycle/src/download_vspec.py:99]: should we also add the other versions? I know it most probably won't happen but for completeness maybe..

File: vehicle-model-lifecycle/src/download_vspec.py [hunk 2]
```diff
@@ -84,21 +87,104 @@ def download_file(uri: str, local_file_path: str):
...
+def get_legacy_model_src(app_manifest_dict: Dict[str, Any]) -> str:
+    """Get the source from the legacy vehicle model (app manifest < v3)
+
+    Args:
+        app_manifest_dict (Dict[str, Any]): The app manifest dict.
 
+    Returns:
+        str: The source URI of the vehicle model
+    """
     possible_keys = ["vehicleModel", "VehicleModel"]
```
Comment (MP91) [inline vehicle-model-lifecycle/src/download_vspec.py:111]:
```suggestion
    possible_keys = ["vehicleModel", "VehicleModel", "vehiclemodel", "Vehiclemodel"]
```
maybe better with an regex? or de we have a rule somewhere for that?

**Comment by MP91** [comment]
another small request since we currently touching this repo anyways: could you remove the pspester.pester-test extension from setup/src/python/common/devcontainer.json? 

We don't have powershell at all and it is currently always showing a notification when starting the devcontainer...

If you disagree, it is also fine, than I create a new PR with the change :)

File: vehicle-model-lifecycle/src/download_vspec.py (hunk 1 above)
Comment (doosuu) [reply vehicle-model-lifecycle/src/download_vspec.py:99]: We don't really have other versions we could distinguish. Maybe we could check the availability of fields to deduct the version  of the manifest?

File: vehicle-model-lifecycle/src/download_vspec.py (hunk 2 above)
Comment (doosuu) [reply vehicle-model-lifecycle/src/download_vspec.py:111]: This is just the 2 different version from app manifest v1 and app manifest v2, any further variants are not supported. Hence I would not add these options or switch to regex.

File: vehicle-model-lifecycle/src/download_vspec.py (hunk 1 above)
Comment (MP91) [reply vehicle-model-lifecycle/src/download_vspec.py:99]: If we now have v3 we do have also v1 and v2 but thats what I meant, most probably nobody will ever write a manifest with manifestVersion: v1...

lets keep it as is and don't waste much time on that.

**Review by MP91 — DISMISSED** [review]
lgtm

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -105,7 +105,15 @@ jobs:
 
       - name: Prepare .velocitas.json for integration test
         run: |
-          NEW_CONFIG="$(jq --arg GITHUB_SHA "$GITHUB_SHA" '.packages[0].version |= $GITHUB_SHA' test/${{ matrix.language }}/.velocitas.json)"
+          COMMIT_REF=${GITHUB_SHA}
+
+          if [[ "$GITHUB_EVENT_NAME" == "pull_request" ]];
+          then
+            echo "Running in context of a PR!"
+            COMMIT_REF=main
```
Comment (MP91) [inline .github/workflows/ci.yml:113]:
```suggestion
            COMMIT_REF=$GITHUB_HEAD_REF
```

**Pull request merged**